webapp/src/main/java/com/box/l10n/mojito/smartling/SmartlingClient.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    HttpHeaders getHttpHeadersForJson() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return headers;
+    }
+
+    HttpHeaders getHttpHeadersForJsonWithAuthorization() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("Authorization", "Bearer " + authToken);
+        return headers;
+    }
+
+    <T> HttpEntity<T> getHttpEntityForPayload(T payload) {
+        return new HttpEntity<>(payload, getHttpHeadersForJson());
+    }
+
+    <T> HttpEntity<T> getHttpEntityForPayloadWithAuthorization(T payload) {
+        return new HttpEntity<>(payload, getHttpHeadersForJsonWithAuthorization());
+    }
+
+    public void authenticate() throws SmartlingClientException {
+        Map<String, Object> payload = getPayloadMapAuthentication();
+        AuthenticationResponse authenticationResponse;
+        try {
+            authenticationResponse = postForAuthentication(payload, AUTH_API_V2_AUTHENTICATE);
+        } catch (SmartlingClientException e) {
+            throw e;
+        }
+        if (authenticationResponse.isAuthenticationErrorResponse()) {
+            throw new SmartlingClientException("Error authenticating with Smartling API, please check your credentials");
+        } else if (!authenticationResponse.isSuccessResponse()) {
+            throw new SmartlingClientException(authenticationResponse.getErrorMessage());
+        } else {
+            Instant now = Instant.now();
+            nextAuthentication = now.plus(Duration.ofHours(authenticationDurationHours));
+            nextRefresh = now.plus(Duration.ofMinutes(refreshDurationMinutes));
+            authToken = authenticationResponse.getResponse().getData().getAccessToken();
+            refreshToken = authenticationResponse.getResponse().getData().getRefreshToken();
+        }
+    }
+
+    AuthenticationResponse postForAuthentication(Map<String, Object> payload, String authApiV2) throws SmartlingClientException {
+        HttpEntity<Map<String, Object>> httpEntity = getHttpEntityForPayload(payload);
+        AuthenticationResponse postForObject = restTemplate.postForObject(
+                authApiV2, httpEntity, AuthenticationResponse.class);
+        return postForObject;
+    }
+
+    public void refresh() throws SmartlingClientException {
+        Instant now = Instant.now();
+        Instant nextCalculatedRefresh = now.plus(Duration.ofMinutes(refreshDurationMinutes));
+        if (now.isAfter(nextAuthentication) || nextCalculatedRefresh.isAfter(nextAuthentication)) {
+            authenticate();
+        } else if (!(nextRefresh.isAfter(now) && nextAuthentication.isAfter(now))) {
+            Map<String, Object> payload = getPayloadMapRefresh();
+            AuthenticationResponse authenticationResponse;
+            try {
+                authenticationResponse = postForAuthentication(payload, AUTH_API_V2_REFRESH);
```

Comment:
same, try catch is not needed since the method `throws SmartlingClientException`